lib/cypress/scoop_and_filter.rb

```diff
@@ -0,0 +1,60 @@
+module Cypress
+  class ScoopAndFilter
+    def initialize(measures)
+      @relevant_codes = codes_in_measures(measures)
+      @demographic_oids = ['2.16.840.1.113883.10.20.28.3.55', '2.16.840.1.113883.10.20.28.3.59', '2.16.840.1.113883.10.20.28.3.56']
+      @hqmf_oids_for_measures = []
+      @hqmf_oids_for_measures = get_all_hqmf_oids_definition_and_status(measures) - @demographic_oids
+    end
+
+    # return an array of all of the concepts in all of the valueset for the measure
+    def codes_in_measures(measures)
+      valuesets = []
+      code_list = []
+      measures.each do |measure|
+        measure['value_set_oid_version_objects'].each do |valueset|
+          valuesets.concat(HealthDataStandards::SVS::ValueSet.where(oid: valueset.oid, version: valueset.version).to_a)
+        end
+      end
+      valuesets.each do |valueset|
+        code_list.concat(valueset.concepts)
+      end
+      code_list.map { |cl| { code: cl.code, codeSystem: cl.code_system_name } }
+    end
+
+    def get_all_hqmf_oids_definition_and_status(measures)
+      all_hqmf_oids = []
+      measures.each do |measure|
```

Comment:
It might be better to replace this whole function with the following:
```
def get_all_hqmf_oids_definition_and_status(measures)
  measures.collect do |measure|
    measure.source_data_criteria.collect do |_key, criteria|
      get_all_hqmf_oids(criteria['definition'], criteria['status'])
    end
  end.flatten.uniq
end
```

I ran a crude speed test and it seems collect is a lot faster overall and it over. Below are the speed test results with the old and new methods:

**Old:**
```
       user     system      total        real
   0.070000   0.000000   0.070000 (  0.072991)

       user     system      total        real
   0.060000   0.000000   0.060000 (  0.061780)

       user     system      total        real
   0.010000   0.000000   0.010000 (  0.014338)
```

**New:**
```
       user     system      total        real
   0.020000   0.000000   0.020000 (  0.032177)

       user     system      total        real
   0.020000   0.000000   0.020000 (  0.015087)

       user     system      total        real
   0.020000   0.000000   0.020000 (  0.017952)
```

Reply:
Also some of these return nil, and that nil seems to get included in the returned information. Is this intended behavior?